From f654bcf1919c011d16a73ff76d6fc517ce7ffef8 Mon Sep 17 00:00:00 2001
From: Michael Forney <mforney@mforney.org>
Date: Mon, 1 Jul 2019 22:45:12 -0700
Subject: [PATCH] Use __typeof__ instead of typeof

typeof might not be available if built with ISO C mode.
---
 util/fusermount.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/fusermount.c b/util/fusermount.c
index ed62ea9..c497891 100644
--- a/util/fusermount.c
+++ b/util/fusermount.c
@@ -992,7 +992,7 @@ static int check_perm(const char **mntp, struct stat *stbuf, int *mountpoint_fd)
 	 * (https://bazaar.launchpad.net/~ecryptfs/ecryptfs/trunk/view/head:/src/utils/mount.ecryptfs_private.c#L225)
 	 * but got expanded as we found more filesystems that needed to be
 	 * overlayed. */
-	typeof(fs_buf.f_type) f_type_whitelist[] = {
+	__typeof__(fs_buf.f_type) f_type_whitelist[] = {
 		0x61756673 /* AUFS_SUPER_MAGIC */,
 		0x00000187 /* AUTOFS_SUPER_MAGIC */,
 		0xCA451A4E /* BCACHEFS_STATFS_MAGIC */,
-- 
2.30.0

